src/main/java/io/logz/jmx2logzio/configuration/Jmx2LogzioConfiguration.java

```diff
@@ -0,0 +1,228 @@
+package io.logz.jmx2logzio.configuration;
+
+import com.typesafe.config.Config;
+import io.logz.jmx2logzio.Jmx2LogzioJavaAgent;
+import io.logz.jmx2logzio.clients.JolokiaClient;
+import io.logz.jmx2logzio.exceptions.IllegalConfiguration;
+import io.logz.jmx2logzio.objects.LogzioJavaSenderParams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.regex.Pattern;
+
+public class Jmx2LogzioConfiguration {
+    private static final Logger logger = LoggerFactory.getLogger(Jmx2LogzioConfiguration.class);
+
+    public static final String LISTENER_URL = "LISTENER_URL";
+    public static final String WHITE_LIST_REGEX = "WHITE_LIST_REGEX";
+    public static final String BLACK_LIST_REGEX = "BLACK_LIST_REGEX";
+    public static final String LOGZIO_TOKEN = "LOGZIO_TOKEN";
+    public static final String SERVICE_NAME = "SERVICE_NAME";
+    public static final String SERVICE_HOST = "SERVICE_HOST";
+    public static final String POLLING_INTERVAL_IN_SEC = "POLLING_INTERVAL_IN_SEC";
+    public static final String FROM_DISK = "FROM_DISK";
+    public static final String IN_MEMORY_QUEUE_CAPACITY = "IN_MEMORY_QUEUE_CAPACITY";
+    public static final String LOGS_COUNT_LIMIT = "LOGS_COUNT_LIMIT";
+    public static final String DISK_SPACE_CHECKS_INTERVAL = "DISK_SPACE_CHECKS_INTERVAL";
+    public static final String QUEUE_DIR = "QUEUE_DIR";
+    public static final String FILE_SYSTEM_SPACE_LIMIT = "FILE_SYSTEM_SPACE_LIMIT";
+    public static final String CLEAN_SENT_METRICS_INTERVAL = "CLEAN_SENT_METRICS_INTERVAL";
+
+    private Pattern whiteListPattern;
+    private Pattern blackListPattern;
+    private String jolokiaFullUrl;
+
+    private LogzioJavaSenderParams logzioJavaSenderParams;
+
+    /* Short name of the sampled service, required = false */
+    private String serviceName;
+
+    /* host of the sampled service */
+    private String serviceHost = null;
+
+    /* Metrics polling interval in seconds */
+    private int metricsPollingIntervalInSeconds = 30;
+
+    // Which client should we use
+    private MetricClientType metricClientType;
+
+    public enum MetricClientType {
+        JOLOKIA,
+        MBEAN_PLATFORM
+    }
+
+    public Jmx2LogzioConfiguration(Config config) throws IllegalConfiguration {
```

Comment:
`Jmx2LogzioConfiguration` too long and hard to follow. Take out to smaller functions..